Description: Fix __testandset on s390x.
 __testandset was broken on s390x, because it assumed 32bit as well
 (the code was identical to s390) - this would cause lockups on spin
 locks.
Author: Christian Seiler <christian@iwakd.de>
Last-Update: 2017-01-05

--- a/s390x/__testandset.S
+++ b/s390x/__testandset.S
@@ -3,8 +3,8 @@
 .type	__testandset,@function
 __testandset:
 	la	%r1,0(%r2)
-	lhi	%r0,1
-	lhi	%r2,0
+	lghi	%r0,1
+	lghi	%r2,0
 	cs	%r2,%r0,0(%r1)
 	xr	%r2,%r0
 	br	%r14
